R/export_tibble.R

```diff
@@ -51,53 +52,93 @@ check_var_type <- function(GADSdat) {
 
 
 ###  add labels to a single variable ---------------------------------------------------------
-addLabels_single <- function(label_df, varClass) {
-  out <- list()
+addLabels_single <- function(varName, label_df, raw_dat, varClass) {
+  attr_list <- list()
   # attributes on variable level
-  if(!all(is.na(label_df[, "varLabel"]))) out[["label"]] <- unique(label_df[, "varLabel"])
-  if(!all(is.na(label_df[, "format"]))) out[["format.spss"]] <- unique(label_df[, "format"])
-  if(!all(is.na(label_df[, "display_width"]))) out[["display_width"]] <- unique(label_df[, "display_width"])
+  if(!all(is.na(label_df[, "varLabel"]))) attr_list[["label"]] <- unique(label_df[, "varLabel"])
+  if(!all(is.na(label_df[, "format"]))) attr_list[["format.spss"]] <- unique(label_df[, "format"])
+  if(!all(is.na(label_df[, "display_width"]))) attr_list[["display_width"]] <- unique(label_df[, "display_width"])
   labeled <- unique(label_df[, "labeled"])
   # check
-  unique_attr <- unlist(lapply(out, length))
+  unique_attr <- unlist(lapply(attr_list, length))
   stopifnot(all(unique_attr)  <= 1)
 
   # missing labels, if any
-  miss_values <- label_df[which(label_df$missings == "miss"), "value"]
-  if(length(miss_values) > 0 && length(miss_values) <= 3)  out[["na_values"]] <- miss_values
-  if(length(miss_values) > 3)  out[["na_range"]] <- range(miss_values)
+  attr_list <- add_miss_tags(varName = varName, attr_list = attr_list, label_df = label_df, raw_dat = raw_dat)
 
-  # out[["class"]] <- strsplit(out[["class"]], split = ", ")[[1]]
   # give specific class depending on a) value labels are needed or not and b) missings are given or not:
-  if(identical(labeled, "yes")) out[["class"]] <- c("haven_labelled")
-  if(identical(labeled, "yes") & all(is.na(label_df$value))) out[["class"]] <- c("haven_labelled_spss")
-  any_miss <- length(miss_values) > 0
-  if(identical(out[["class"]], "haven_labelled") && any_miss) out[["class"]] <- c("haven_labelled_spss", "haven_labelled")
+  if(identical(labeled, "yes")) attr_list[["class"]] <- c("haven_labelled")
+  if(identical(labeled, "yes") & all(is.na(label_df$value))) attr_list[["class"]] <- c("haven_labelled_spss")
+  any_miss <- !is.null(attr_list$na_values) || !is.null(attr_list$na_range)
+  if(identical(attr_list[["class"]], "haven_labelled") && any_miss) attr_list[["class"]] <- c("haven_labelled_spss", "haven_labelled")

```

Comment:
I agree, this was written poorly. I have a strong tendency to avoid `else` if possible. However, I think I have adopted most of your suggestions (with some slight modifications).